data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/tekuv1/beacon/GetProposersData.java

```diff
@@ -13,38 +13,51 @@
+import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_OK;
 import static tech.pegasys.teku.infrastructure.http.RestApiConstants.CACHE_NONE;
 import static tech.pegasys.teku.infrastructure.http.RestApiConstants.RES_INTERNAL_ERROR;
 import static tech.pegasys.teku.infrastructure.http.RestApiConstants.RES_OK;
 import static tech.pegasys.teku.infrastructure.http.RestApiConstants.RES_SERVICE_UNAVAILABLE;
 import static tech.pegasys.teku.infrastructure.http.RestApiConstants.SERVICE_UNAVAILABLE;
 import static tech.pegasys.teku.infrastructure.http.RestApiConstants.TAG_EXPERIMENTAL;
 
-import io.javalin.core.util.Header;
+import com.fasterxml.jackson.core.JsonProcessingException;
 import io.javalin.http.Context;
-import io.javalin.http.Handler;
 import io.javalin.plugin.openapi.annotations.HttpMethod;
 import io.javalin.plugin.openapi.annotations.OpenApi;
 import io.javalin.plugin.openapi.annotations.OpenApiContent;
 import io.javalin.plugin.openapi.annotations.OpenApiResponse;
 import tech.pegasys.teku.api.DataProvider;
 import tech.pegasys.teku.api.NodeDataProvider;
 import tech.pegasys.teku.api.response.v1.teku.GetProposersDataResponse;
-import tech.pegasys.teku.provider.JsonProvider;
-
-public class GetProposersData implements Handler {
+import tech.pegasys.teku.beaconrestapi.MigratingEndpointAdapter;
+import tech.pegasys.teku.beaconrestapi.schema.ProposersData;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.EndpointMetadata;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiRequest;
 
+public class GetProposersData extends MigratingEndpointAdapter {
   public static final String ROUTE = "/teku/v1/beacon/proposers_data";
 
-  private final JsonProvider jsonProvider;
   private final NodeDataProvider nodeDataProvider;
 
-  public GetProposersData(final DataProvider provider, final JsonProvider jsonProvider) {
-    this(provider.getNodeDataProvider(), jsonProvider);
+  public GetProposersData(final DataProvider provider) {
+    this(provider.getNodeDataProvider());
   }
 
-  GetProposersData(final NodeDataProvider nodeDataProvider, final JsonProvider jsonProvider) {
-    this.jsonProvider = jsonProvider;
+  GetProposersData(final NodeDataProvider nodeDataProvider) {
+    super(
+        EndpointMetadata.get(ROUTE)
+            .operationId("getProposersData")
+            .summary("Get current prepared beacon proposers and registered validators")
+            .description(
+                "Get the current proposers information held by beacon node as result of "
+                    + "prepare_beacon_proposer and register_validator validator API calls. This API is "
+                    + "considered unstable and the returned data format may change in the future.")
+            .tags(TAG_EXPERIMENTAL)
+            .response(SC_OK, "Request successful", ProposersData.getJsonTypeDefinition())
+            .withServiceUnavailableResponse() // TODO check if needed
```

Comment:
its documented, so i'd probably leave it, but if you want to see if its likely, could start syncing from a state thats a day or 2 old and then call this endpoint...